src/diskquota_utility.c

```diff
@@ -1683,3 +1683,25 @@ DiskquotaShmemInitHash(const char           *name,       /* table string name fo
 	return ShmemInitHash(name, init_size, max_size, infoP, hash_flags | HASH_BLOBS);
 #endif /* GP_VERSION_NUM */
 }
+
+/*
+ * Returns HASH_FIND if hash table is full and HASH_ENTER otherwise.
+ */
+HASHACTION
+check_hash_fullness(HTAB *hashp, int max_size, const char *warning_message, TimestampTz *last_overflow_report,
+                    int guc_value)
+{
+	if (hash_get_num_entries(hashp) < max_size) return HASH_ENTER;
+
+	if (hash_get_num_entries(hashp) == max_size)
+	{
+		TimestampTz current_time = GetCurrentTimestamp();
+		if (TimestampDifferenceExceeds(*last_overflow_report, current_time,
+		                               diskquota_hashmap_overflow_report_timeout * 1000))
+		{
+			ereport(WARNING, (errmsg(warning_message, guc_value)));
```

Comment:
remove `%d` from warning_message and
```suggestion
			ereport(WARNING, (errmsg("%s %d", warning_message, guc_value)));
```

Reply:
But what does it change?

Reply:
with this change warning_message is not string format template

Reply:
I don't see any improvements except that now the GUC value will always be at the end.

Reply:
The current version of the check_hash_fullness function can be passed a string with multiple substitutions to format the message, which will result in a runtime error.